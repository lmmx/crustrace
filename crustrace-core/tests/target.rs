use crustrace_core::instrument_impl;
use insta::assert_snapshot;
use proc_macro2::TokenStream;
use quote::quote;
use rust_format::{Formatter, RustFmt};

fn apply_instrument(args: TokenStream, input: TokenStream) -> String {
    let output = instrument_impl(args, input).expect("Should instrument successfully");
    println!("Instrumented: {}", output);
    let fmt_str = RustFmt::default()
        .format_tokens(output)
        .unwrap_or_else(|e| panic!("Format error: {}", e));
    println!("Formatted: {}", fmt_str);
    fmt_str
}

#[test]
fn test_target_parsing() {
    let args = quote!(target = "my_crate::my_target");
    let item = quote! {
        fn test_function(x: u32) -> u32 {
            x + 1
        }
    };

    assert_snapshot!(apply_instrument(args, item));
}
