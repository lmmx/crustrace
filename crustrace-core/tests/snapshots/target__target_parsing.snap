---
source: crustrace-core/tests/target.rs
expression: "apply_instrument(args, item)"
---
fn test_function(x: u32) -> u32 {
    let __tracing_attr_span = tracing :: span ! (target : "my_crate::my_target" , tracing :: Level :: INFO , "test_function" , x = x);
    let __tracing_attr_guard = __tracing_attr_span.enter();
    {
        x + 1
    }
}
